### ISSUE TITLE
Add option to ignore property names in `id-length` rule

### ISSUE THREAD

**Issue by mathieumg** [eslint/eslint#3450]
> Requesting a feature? Please be sure to include:
>
>The problem you want to solve (don't mention the solution)
>Your take on the correct solution to problem

I would like to suggest adding the `properties` configuration to the `id-length` rule such that property names are ignored, exactly like it's currently possible to do in http://eslint.org/docs/rules/camelcase#options

The rationale behind this is that object properties often have enough context to be shorter or are imposed by third-party interfaces (`url` and `src` come to mind), whereas other identifiers like variables and functions require more characters to be self descriptive. To follow with that example, I use `id-length` with a value of `4` and I could see my code having the following:

```js
const myObj = {
  src: "...",
  url: "...,
};
```

but not the following:

```js
const src = "...";
const url = "...";
```

I hope I provided enough information, don't hesitate to ask for more if anything is missing, thank you!

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by BYK** [comment]
Currently, if you quote the properties, they are not checked (this was intentional). Would this solve your problem?

**Comment by mathieumg** [comment]
While it's good to know as a workaround, it's not enough. We also use `"quote-props": [2, "as-needed"]`.

**Comment by BYK** [comment]
Fair point. I don't want to complicate the rule too much but this seems like an okay improvement. Any thoguhts @eslint/eslint-team?

**Comment by nzakas** [comment]
Seems reasonable to me.
